channeldb/payment_control.go

```diff
@@ -95,12 +95,20 @@ type PaymentControl struct {
 	currPaymentSeq   uint64
 	storedPaymentSeq uint64
 	db               *DB
+	cfg              *PaymentControlConfig
+}
+
+// PaymentControlConfig houses necessary fields that a PaymentControl needs to
+// function.
+type PaymentControlConfig struct {
+	KeepFailedPayments bool
 }
 
 // NewPaymentControl creates a new instance of the PaymentControl.
-func NewPaymentControl(db *DB) *PaymentControl {
+func NewPaymentControl(db *DB, cfg *PaymentControlConfig) *PaymentControl {
```

Comment:
The config probably doesn't belong here. Please check `config_builder.go` re [how we set options for db](https://github.com/lightningnetwork/lnd/blob/master/config_builder.go#L850). Since it's db-related config, it makes sense to add the config option there, and access it via the `db` instance here.

Reply:
Fixed!